src/compiler/types.ts

```diff
@@ -2365,6 +2365,11 @@ namespace ts {
         sourceFiles: SourceFile[];
     }
 
+    export interface JsonSourceFile extends SourceFile {
+        jsonObject?: ObjectLiteralExpression;

```

Comment:
While it makes sense for parsing _tsconfig.json_, JSON isn't necessarily always an object literal. In the lib reference PR, I started using [`ts.parseConfigFileTextToJson`](https://github.com/Microsoft/TypeScript/pull/15780/files#diff-0994635726d7ae5036c62009690e571aR71) to parse a single _libs.json_ file shared between the gulpfile and jakefile. I'd have to make some changes since [_libs.json_ has an array as its root](https://github.com/Microsoft/TypeScript/pull/15780/files#diff-15c93c9538c61c5e506a49ecefcbba8e).